app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :post, inverse_of: :comments, foreign_key: :post_id, counter_cache: true
+  belongs_to :user, inverse_of: :comments, foreign_key: :user_id, counter_cache: true
+
+  # update comments counter
+
+  def update_comment_counter
+    post.update(comment_counter: post.comments.size)
+  end
```

Comment:
- [ ] You should increment the comment counter for the post in this method and call the method after saving so when a user creates a comment the counter should be incremented automatically. Please fix this issue.
  - You can consider the below code.
```ruby
after_save :update_comment_counter
  def update_comment_counter
    post.increment!(:comment_counter)
  end